sui/upgrade-gateway.js

```diff
@@ -0,0 +1,93 @@
+const { Command, Option } = require('commander');
+const { TxBuilder } = require('@axelar-network/axelar-cgp-sui');
+const { bcs } = require('@mysten/sui.js/bcs');
+const { fromB64, toB64 } = require('@mysten/bcs');
+const { addBaseOptions } = require('./cli-utils');
+const { getWallet, broadcast } = require('./sign-utils');
+const { loadSuiConfig } = require('./utils');
+const { saveConfig, printInfo, validateParameters, writeJSON } = require('../evm/utils');
+
+async function processCommand(chain, options) {
+    const [keypair, client] = getWallet(chain, options);
+    printInfo('Wallet address', keypair.toSuiAddress());
+
+    const { offline, policy, sender, txFilePath } = options;
+    const gatewayConfig = chain.contracts.axelar_gateway ?? {};
+
+    const builder = new TxBuilder(client);
+    const { modules, dependencies, digest } = await builder.getContractBuild('axelar_gateway');
+    const upgradeCap = options.upgradeCap || gatewayConfig.objects?.UpgradeCap;
+    const digestHash = options.digest ? fromB64(options.digest) : digest;
+    const packageId = gatewayConfig.address;
+
+    validateParameters({
+        isNonEmptyString: { policy },
+        isNonEmptyStringArray: { modules, dependencies },
+        isKeccak256Has: { upgradeCap, packageId },
+    });
+
+    const tx = builder.tx;
+    const cap = tx.object(upgradeCap);
+
+    const ticket = tx.moveCall({
+        target: '0x2::package::authorize_upgrade',
+        arguments: [cap, tx.pure(policy), tx.pure(bcs.vector(bcs.u8()).serialize(digestHash).toBytes())],
+    });
+
+    const receipt = tx.upgrade({
+        modules,
+        dependencies,
+        packageId,
+        ticket,
+    });
+
+    tx.moveCall({
+        target: '0x2::package::commit_upgrade',
+        arguments: [cap, receipt],
+    });
+
+    if (offline) {
+        sender ? tx.setSender(sender) : tx.setSender(keypair.toSuiAddress());
```

Comment:
this conflicts with Foivos's PR where helpers have been created for upgrading. we should use that since it'll be a common pattern between contracts